### ISSUE TITLE
Cube2: Sauerbraten

### ISSUE THREAD

**Issue by 2-bithacker** [Botspot/pi-apps#1475]
### What is the name of the app?

Cube2: Sauerbraten

### (Optional) Where is the app hosted?

http://sauerbaten.org/

### About the app

Cube2: Sauerbraten is a fast-paced, open source first person shooter based on the cube2 game engine.  It runs decently on the raspi4 (30-150 fps on low detail) And is multiplayer.  It resembles quake in most ways, but has an in-game map editor and is a lot funner than quake.    It uses about 1.2 gb of space when installed.  Now this app can be installed with apt (sudo apt install sauerbraten) but in the raspbian release buster apt repo and older, they have the old, dead version.  My Install script downloads the latest version and then builds it from source.   I think this game would fill in a gap in the pi-apps software list.  The list has has a first person shooter, doom3, but that is only campaign.  This fps is much less popular than Id software games, due to the lack of advertizing, but it is suprisingly clean and well-designed.  I bombed the last app attempt, but with this one I came ready.  I already have a complete pi-app. Also, I used to use my rpi4 to play sauer before I got a better pc.

### Upload file
[sauerbraten_pi-apps.zip](https://github.com/Botspot/pi-apps/files/8093126/sauerbraten_pi-apps.zip)

here is the complete pi-app.

- [X] I have confirmed that this app is legal and not piracy.
- [X] I have confirmed that this app has never been discussed in https://github.com/Botspot/pi-apps/issues and https://github.com/Botspot/pi-apps/pulls, and it is not in the Pi Apps app list.
- [X] I have confirmed that this app can run on Raspberry Pi.

**Comment by ThaPg** [comment]
If you want, you can modify the App Hosted. http://sauerbraten.org/ is right (you lose "r")

**Comment by 2-bithacker** [comment]
I do not understand.  The app runs ok on raspberry pi, and works better with a special script.

**Comment by ThaPg** [comment]
My fault!!!! I'm sorry!!!!

Where is the app hosted? 
You added wrong webpage. I wrote you before to change it. 

Also. I'm trying to run your script!!!! Thanks!!!

**Comment by 2-bithacker** [comment]
it is hosted at sauerbraten.org!  When you launch it, it says "looking up master sauerbraten.org"

**Comment by ThaPg** [comment]
> ### (Optional) Where is the app hosted?
> sauerbaten.org
> 
Change to http://sauerbraten.org/

**Comment by 2-bithacker** [comment]
oki ty

**Comment by ThaPg** [comment]
The game installed successfully (using your installing script) but i can't find it in game menu or in desktop. Any idea where can I find it?

**Comment by 2-bithacker** [comment]
Yes.  The script is incomplete as I don't know how to make a launcher.  You can launch it by running  `cd ~/pi-apps/apps/Cube2Sauerbraten/sauerbraten`, or where ever you installed it, and then running `./sauerbraten_unix`.  I recommend turning it all to minimum detail ESPECIALLY the option called shaders because that can cut ur fps by 90 percent.

**Comment by ThaPg** [comment]
I have raspberry pi 4 with 4gb ram.

Using your install script and pi-apps settings "Import App" I made install script giving credits directly to you. But if you make a launcher file and desktop icon, I'll stop my work. Do you want to upload .zip file to continue the work?

**Comment by 2-bithacker** [comment]
https://sourceforge.net/projects/sauerbraten/files/sauerbraten/2020_11_29/sauerbraten_2020_12_29_linux.tar.bz2/download  click that.  That will download the zip I would have uploaded.  I didn't make that app! It was made by many people.

**Comment by 2-bithacker** [comment]
There.  I have completed the app.  Launch script, icons, install script, remove script, etc.  Put the folder in /home/pi/pi-apps/apps and you'll be all set!/home/optiplex780/Documents/Cube2Sauerbraten.zip

**Comment by 2-bithacker** [comment]
> Using your install script and pi-apps settings "Import App" I made install script giving credits directly to you. But if you make a launcher file and desktop icon, I'll stop my work. Do you want to upload .zip file to continue the work?

I made a  better install script and a launcher file.  What work are you doing, and why will you stop it if I make a launcher file and Icon?

**Comment by 2-bithacker** [comment]
[sauerbraten_pi-apps.zip](https://github.com/Botspot/pi-apps/files/8093108/sauerbraten_pi-apps.zip)
Here is the complete pi-app.

**Comment by Botspot** [comment]
> [sauerbraten_pi-apps.zip](https://github.com/Botspot/pi-apps/files/8093108/sauerbraten_pi-apps.zip) Here is the complete pi-app, ready for distribution. It has ALL the scripts to download, install, lauch, and remove cube2 sauerbraten. I tested it and it works.

I apologise for taking so long.
Your script has a few problems.
```bash
#!/bin/bash


#Be sure to use the "error" function - it will display a message if a command fails to run. Example below:
rm ~/pi-apps/apps/Cube2Sauerbraten/sauerbraten.init &>/dev/null
rm ~/Desktop/sauerbraten.init &>/dev/null
rm ~/pi-apps/apps/Cube2Sauerbraten/sauerbraten.init &>/dev/null
rm download &>/dev/null
rm ~/sauerbraten &>/dev/null
sudo apt update || warning
sudo apt install --yes zlib1g-dev libgeoip-dev build-essential libsdl2-dev libsdl2-mixer-dev libsdl2-image-dev || error 'Install of sdl2 and opengl libraries failed!'
wget https://sourceforge.net/projects/sauerbraten/files/sauerbraten/2020_11_29/sauerbraten_2020_12_29_linux.tar.bz2/download || error 'Download failed from sourceforge.net!'
tar -xf download || error 'extract of Cube2: Sauerbraten failed!'
make install -C ~/sauerbraten/src|| error 'build failed!'
rm download &>/dev/null
cp ~/pi-apps/apps/Cube2Sauerbraten/launcher.init ~/pi-apps/apps/Cube2Sauerbraten/sauerbraten.init
mv ~/pi-apps/apps/Cube2Sauerbraten/sauerbraten.init ~/Desktop
chmod +x ~/Desktop/sauerbraten.init
chmod +x ~/sauerbraten/bin_unix/native_server
chmod +x ~/sauerbraten/bin_unix/native_client
# Install some packages that are necessary to run this app - no need for "error", as the install_packages function already handles errors.
#exit 0
```
First of all, why are you deleting `~/pi-apps/apps/Cube2Sauerbraten/sauerbraten.init` *twice*, and then trying to copy it somewhere? For that matter, why are you handling any files within the `pi-apps` directory at all?
Also, you are not installing packages correctly. Instead of `sudo apt`, you should be using the `install_packages` function. Please read [the app-creation tutorial](https://github.com/Botspot/pi-apps/wiki/Creating-an-app), and if you have any questions please ask them.

**Comment by 2-bithacker** [comment]
I deleted it sauerbraten.init twice as a cleanup thing so someone could run the install script again without having to uninstall.  sauerbraten.init is the desktop launcher. It makes a copy of the one in the pi-apps directory and moves it to the desktop.   I had the sudo apt part because this was actually a script that I had made just to install sauerbraten when I realized it  would work on raspberry pi with a couple mods.
